### PR TITLE
Use `std::string::compare` and `operator==` where possible

### PR DESCRIPTION
**Pull request by gaul**

### CODE REVIEW

File: src/cache.cpp [hunk 1]
```diff
@@ -248,7 +248,7 @@ bool StatCache::GetStat(const std::string& key, struct stat* pst, headers_t* met
                     std::string tag = lower(hiter->first);
                     if(tag == "etag"){
                         stretag = hiter->second;
-                        if('\0' != petag[0] && 0 != strcmp(petag, stretag.c_str())){
+                        if('\0' != petag[0] && petag != stretag){
```
Comment (ggtakec) [inline src/cache.cpp:251]: By the way, why don't you make the following changes?
`'\0' != petag[0]` -> `!petag.empty()`

**Review by ggtakec — COMMENTED** [review]
Please check a comment.

File: src/cache.cpp (hunk 1 above)
Comment (gaul) [reply src/cache.cpp:251]: `petag` is a `char*`.

Comment (ggtakec) [reply src/cache.cpp:251]: Sorry for my fault.

**Review by ggtakec — APPROVED** [review]

**Pull request merged**